test/e2e/operator/deployment_api.go

```diff
@@ -144,6 +161,30 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 			})
 		}
 
+		It("get deployment shall list expected fields", func() {
+			expectedFiles := len([]string{"NAME", "DEVICEMODE", "NODESELECTOR", "IMAGE", "STATUS", "AGE"})
+			deployment := getDeployment("test-get-deployment-fields")
+			// Only values that are visible in Deployment CR are shown in `kubectl get`
+			// but, not the default values chosen by the operator.
+			// So provide the values that are expected to list.
+			deployment.Spec.DeviceMode = api.DeviceModeDirect
+			deployment.Spec.PullPolicy = corev1.PullNever
+			deployment.Spec.Image = dummyImage
+			deployment.Spec.NodeSelector = map[string]string{
+				"storage": "unknown-node",
+			}
+
+			deployment = deploy.CreateDeploymentCR(f, deployment)
+			defer deploy.DeleteDeploymentCR(f, deployment.Name)
+
+			// Run in-cluster kubectl from master node
+			ssh := os.Getenv("REPO_ROOT") + "/_work/" + os.Getenv("CLUSTER") + "/ssh.0"
+			out, err := exec.RunCommand(ssh, "kubectl", "get", "deployments.pmem-csi.intel.com", "--no-headers")
+			Expect(err).ShouldNot(HaveOccurred(), "kubectl get: %v", out)
+			fields := strings.Fields(string(out))
+			Expect(len(fields)).Should(BeEquivalentTo(expectedFiles), "fields mismatch: %v", out)
```

Comment:
The new test isn't passing:
```
/mnt/workspace/pmem-csi_PR-749/test/e2e/operator/deployment_api.go:164
Nov 1 03:50:34.611: fields mismatch: test-get-deployment-fields direct map[storage:unknown-node] unexisting/pmem-csi-driver 1s
Expected
<int>: 5
to be equivalent to
<int>: 6
/mnt/workspace/pmem-csi_PR-749/test/e2e/operator/deployment_api.go:185
```

Reply:
`MatchRegexp` (https://github.com/onsi/gomega/blob/v1.10.3/matchers.go#L171) might be a better way to check for the expected output.

Reply:
> The new test isn't passing:

Fixed it now.

Reply:
> `MatchRegexp` (https://github.com/onsi/gomega/blob/v1.10.3/matchers.go#L171) might be a better way to check for the expected output.

Changed the test to use regex.

Reply:
This is a bit dangerous: it only works if the strings that you put into the regex have no special meaning. I'm on the edge whether the curly brackets in `{"storage":"unknown-node"}` are matcher literally or might be interpreted as a repetition count. But it seems to work, so I suppose it's okay....

... except that the output with Kubernetes 1.17 is different:
https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/PR-749/20/tests

```
Expected
<string>: test-get-deployment-fields direct map[storage:unknown-node] unexisting/pmem-csi-driver Running 14s
to match regular expression
<string>: test-get-deployment-fields\s+direct\s+{"storage":"unknown-node"}\s+unexisting/pmem-csi-driver\s+Running\s+[0-9]+(s|m)
/mnt/workspace/pmem-csi_PR-749/test/e2e/operator/deployment_api.go:188
```

Reply:
Made changes to regex to support K8s 1.17 output., this also avoids usage of `{}` in the regex.